core/include/vecmem/memory/memory_resource.hpp

```diff
@@ -38,3 +38,54 @@ using memory_resource = std::experimental::pmr::memory_resource;
 #else
 #error "C++17 LFTS V1 (P0220R1) component memory_resource not found!?!"
 #endif
+
+#if defined(VECMEM_NOT_HAVE_DEFAULT_RESOURCE)
+
+#include <atomic>
+
+#if defined(VECMEM_HAVE_PMR_MEMORY_RESOURCE)
+namespace std::pmr {
+#else
+namespace std::experimental::pmr {
+#endif
+
+namespace detail {
+class new_delete_resource_impl : public memory_resource {
+    virtual void* do_allocate(std::size_t bytes, std::size_t) override {
+        return malloc(bytes);
+    }
+
+    virtual void do_deallocate(void* p, std::size_t, std::size_t) override {
+        free(p);
+    }
+
+    virtual bool do_is_equal(
+        const memory_resource& other) const noexcept override {
+        return &other == this;
+    }
+};
```

Comment:
`host_memory_resource` depends on that header, also the spec says that `new_delete_resource` is singleton and `p->is_equal(r)` returns `&r == p`, rather than the fancy stuff we do in `host_memory_resource`.

Reply:
We can update `vecmem::host_memory_resource`. Though the singleton stuff I'm not too sure about... :frowning:

Reply:
As said in the other comment, up to you if you want me to update `vecmem::host_memory_resource::do_equal` and get rid of `new_delete_resource_impl`.